app/jobs/discord/handle_interaction_job.rb

```diff
@@ -0,0 +1,231 @@
+      @current_user = User.find_by(discord_id: @user_id) if @user_id
+      @current_event = Event.find_by(discord_guild_id: @guild_id) if @guild_id
+
+      command_name = @interaction.dig(:data, :name)
+
+      unless command_name.in?(::Discord::RegisterCommandsJob.commands.pluck(:name))
+        respond content: "Unknown command: #{command_name}" and return
+      end
+
+      return send("#{command_name.gsub("-", "_")}_command")
+    rescue => e
+      if Rails.env.development?
+        backtrace = e.backtrace.join("\n")
+        if backtrace.length > 4_000
+          backtrace = "#{backtrace[0..4_000]}..."
+        end
+
+        respond content: "**That didn't work!**\nWe're going to debug what went wrong.", embeds: [
+          {
+            title: e.message,
+            description: "```\n#{backtrace}\n```",
+            color: 0xCC0100,
+          }
+        ]
+      else
+        respond content: "**That didn't work!**\nWe're going to debug what went wrong."
+      end
+
+      Rails.error.report(e)
+    end
+
+    private
+
+    def ping_command
+      respond content: "Pong!"
+    end
+
+    def link_command
+      if @current_event.present? && @current_user.present?
+        respond content: "HCB has already been setup for this Discord server!", embeds: linking_embed
+      elsif !@current_event.present? && @current_user.present?
+        respond content: "You've linked your Discord and HCB accounts, but this Discord server isn't connected to an HCB organization yet:", components: button_to("Set up HCB on this server", url_helpers.discord_setup_url(guild_id: @guild_id, channel_id: @channel_id)), embeds: linking_embed
+      elsif @current_event.present? && !@current_user.present?
+        respond content: "HCB has already been setup for this Discord server, but your Discord account isn't linked to your HCB account yet:", components: button_to("Link Discord account", url_helpers.discord_link_url(discord_id: @user_id)), embeds: linking_embed
+      else
+        respond content: "Link your HCB account, and then connect this Discord server to an HCB organization:", components: [button_to("Link Discord account", url_helpers.discord_link_url(discord_id: @user_id)), button_to("Set up HCB on this server", url_helpers.discord_setup_url(guild_id: @guild_id, channel_id: @channel_id))], embeds: linking_embed
+      end
+    end
+
+    def setup_command
+      link_command
+    end
+
+    def balance_command
+      return require_linked_event unless @current_event
+
+      respond embeds: [
+        {
+          title: "#{@current_event.name}'s balance is #{ApplicationController.helpers.render_money @current_event.balance_available_v2_cents}",
+          color:
+        }
+      ], components: button_to("View on HCB", url_helpers.my_inbox_url)
+    end
+
+    def transactions_command
+      return require_linked_event unless @current_event
+
+      transactions = @current_event.canonical_pending_transactions.order(created_at: :desc).limit(4)
```

Comment:
We should probably use this to get the link of transactions:

https://github.com/hackclub/hcb/blob/ca3cecee49ca389b1d652e7613c204804bfaecc2/app/controllers/events_controller.rb#L189-L203

This is because CPT is incomplete and will miss some transactions

Reply:
done